### PR TITLE
Don't enable use-after-return checking on Apple's Clang

### PR DESCRIPTION
**Pull request by jkoritzinsky**
We're not compatible with this validation yet and it looks like Apple's Clang is now defaulting to it being enabled now.

### CODE REVIEW

File: eng/native/configurecompiler.cmake [hunk 1]
```diff
@@ -211,8 +211,8 @@ if (CLR_CMAKE_ENABLE_SANITIZERS)
       # Disable the use-after-return check for ASAN on Clang. This is because we have a lot of code that
```
Comment (danmoseley) [inline eng/native/configurecompiler.cmake:211]: Should this comment still say CLang

Comment (jkoritzinsky) [reply eng/native/configurecompiler.cmake:211]: Yep. Throughout the CMake logic, we refer to both Clang variants as Clang. We don't refer to AppleClang specifically unless we need to do something specific for it (which we haven't so far).

**Comment by jkoritzinsky** [comment]
/ba-g failures are all Helix infra failures and a helix timeout that's not easy to make a Known Issue for.

File: eng/native/configurecompiler.cmake [hunk 2]
```diff
@@ -211,8 +211,8 @@ if (CLR_CMAKE_ENABLE_SANITIZERS)
...
       # depends on the fact that our locals are not saved in a parallel stack, so we can't enable this today.
       # If we ever have a way to detect a parallel stack and track its bounds, we can re-enable this check.
-      add_compile_options($<$<COMPILE_LANG_AND_ID:C,Clang>:-fsanitize-address-use-after-return=never>)
-      add_compile_options($<$<COMPILE_LANG_AND_ID:CXX,Clang>:-fsanitize-address-use-after-return=never>)
+      add_compile_options($<$<COMPILE_LANG_AND_ID:C,AppleClang>:-fsanitize-address-use-after-return=never>)
+      add_compile_options($<$<COMPILE_LANG_AND_ID:CXX,AppleClang>:-fsanitize-address-use-after-return=never>)
```
Comment (am11) [inline eng/native/configurecompiler.cmake:215]: Should we append it or is replacement intentional? i.e.
```suggestion
      add_compile_options($<$<COMPILE_LANG_AND_ID:C,Clang,AppleClang>:-fsanitize-address-use-after-return=never>)
      add_compile_options($<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-fsanitize-address-use-after-return=never>)
```

**Review by am11 — APPROVED** [review]
👍

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**